MapboxSearch/sdk/src/main/java/com/mapbox/search/SearchEngineSettings.kt

```diff
@@ -22,7 +21,7 @@ public class SearchEngineSettings @JvmOverloads constructor(
      * Note that this class requires [Manifest.permission.ACCESS_COARSE_LOCATION] or
      * [Manifest.permission.ACCESS_FINE_LOCATION] to work properly.
      */
-    public val locationEngine: LocationEngine = defaultLocationEngine(),
+    public val locationProvider: DeviceLocationProvider = defaultDeviceLocationProvider(),
```

Comment:
It's better to reference base `com.mapbox.common.location.LocationProvider` interface if you don't need anything specific from `DeviceLocationProvider` in your implementation. Otherwise, everyone who wants to provide own Location Provider implementation will have to deal with unused methods.

This comment is applicable to all the classes that accept location provider as a parameter.

Reply:
Done